src/devices/BuildHat/Brick.cs

```diff
@@ -0,0 +1,1564 @@
+                                                    // Proximity mode
+                                                    if (_sensorType[port] == SensorType.ColourAndDistanceSensor)
+                                                    {
+                                                        ((ColorAndDistanceSensor)_elements[port]).Distance = Convert.ToInt32(elements[inc++]);
+                                                    }
+
+                                                    break;
+                                                case 2:
+                                                    // Counter mode
+                                                    if (_sensorType[port] == SensorType.ColourAndDistanceSensor)
+                                                    {
+                                                        ((ColorAndDistanceSensor)_elements[port]).Counter = Convert.ToInt32(elements[inc++]);
+                                                    }
+
+                                                    break;
+                                                case 3:
+                                                    // reflection mode
+                                                    color.ReflectedLight = Convert.ToInt32(elements[inc++]);
+
+                                                    break;
+                                                case 4:
+                                                    // Ambiant mode
+                                                    color.AmbiantLight = Convert.ToInt32(elements[inc++]);
+
+                                                    break;
+                                            }
+
+                                            break;
+                                        case SensorType.SpikePrimeUltrasonicDistanceSensor:
+                                            var dist = (UltrasonicDistanceSensor)_elements[port];
+                                            dist.Distance = Convert.ToInt32(elements[inc++]);
+                                            break;
+                                        case SensorType.SpikePrimeForceSensor:
+                                            var force = (ForceSensor)_elements[port];
+                                            force.Force = Convert.ToInt32(elements[inc++]);
+                                            force.IsPressed = elements[inc++] == "1";
+                                            break;
+                                        case SensorType.SpikeEssential3x3ColorLightMatrix:
+                                        // nothing to report
+                                        default:
+                                            break;
+                                    }
+                                }
+
+                            }
+                            catch (Exception)
+                            {
+                                // Just skip, it can be malformed
```

Comment:
Maybe add logging?

Reply:
That's not what I meant, actually. We have a logging framework in place. 

In the constructor, do (you might need to add a reference to Iot.Device.Common):
```
_logger = this.GetCurrentClassLogger()
```

Then you can do (without any #if's): 
```
_logger.LogError(ex, $"There was an error parsing the command reply: {ex.Message}");
```

Reply:
Fair, I can add the logging. Will do!